core/metacling/src/TCling.cxx

```diff
@@ -5877,7 +5877,8 @@ static void* LazyFunctionCreatorAutoloadForModule(const std::string& mangled_nam
 
    static bool sFirstRun = true;
    // We don't want to do directory_iterator several times because it contains HDD access
-   static std::set<std::string> sLibraries;
+   static std::vector< std::pair<uint32_t, std::string> > sLibraries;
```

Comment:
Why not size_t?

Reply:
Because size_t is 64 bit and uint32_t is 32 bit, I think path are less than 2**32

Reply:
I see. Is `sLibraries` actually supposed to contain all library candidates for a `path`? If yes, please consider renaming `sAvailableLibraries` or similar.

Reply:
It's not hashing, I'm just saving memory by storing uint32_t rather than path like "mnt/build/jenkins/workspace/root-pullrequests-build/build/"